CONTRIBUTING.md

```diff
@@ -1,4 +1,44 @@
+
+A fourth potential main area is the serializers folder which will be used to build the response.
+
+There is one more file to be familiar with and that is the `app.py` file. This is the file that is ran when the application starts. This is also where we are describing the routes for the app.
+
+The rest of this section will describe the three main areas, and how each of those areas should be tested.
+
+Models:
+
+Models define the database schema. They define what tables to create and what columns to use. They can also contain other methods that relates to the business logic of the application. All models in this application inherit from the BaseModel class, which adds created_at and udpated_at timestamps for all the tables in the database. It also contains methods that are used to find, create, update, and delete database rows. As of this writing there are some models that still have an init method. However, for most of the models the init method is not needed and will be removed. This is because Flask-restful provides an init method that works with key word arguments, and it is recommended to call super if a custom init method is needed. You can see an example of this in the User Model. Currently, all models except for the lease model have a json method that defines how to serialize that object. Models can be found in the `models` directory.
+
+Testing Models:
+
+All models should have unit tests that can be found in the `tests/unit` directory. Each model should have tests that test the inherited methods from the BaseModel to ensure that nothing crazy is inadvertently done to change the behavior of the methods that the BaseModel provides. These are easily implemented using the base_interface_test file. Finally, there should be a test for every public method that is defined in the model. This would be the json method or any other method defined in the file that is directly used outside of the Models class.
+
+Schemas:
+
+Marshmallow Schemas are used primarily for input validation and deserialization. And eventually serialization. Schemas receive the data that the client sends the back-end and validates that data before the data is inserted into the database. They can also describe how the data is serialized before sending data to the client. Schemas can be found in the `schemas` directory.
+
+Testing Schemas:
+
+All schemas should have unit tests. Primarily validation tests. This app uses flask-marshmallow which provides an auto-schema that infers some basic validations based off of the table definition in the Models class. Any additional validations defined in the schema must be tested Schema tests can be found in the `tests/schemas` directory. Deserialization should also be tested here when used. Serialization is currently not used and does not need to be tested here at this time. It's possible that testing serialization will take place elsewhere.
```

Comment:
"Any additional validations defined in the schema must be tested. Schema tests can be found in the `tests/schemas` directory."
(add a period after tested)